### DISCUSSION TITLE
Action modal cancel button

### DISCUSSION THREAD

**Discussion by divdax** [filamentphp/filament#8314] [Help]
Hey! Is it possible to modify the cancel button in the action modal? If I click on cancel, I would like it to trigger something.

```php
return Action::make('move')
    ->requiresConfirmation()
    ->modalCancelAction(fn() => dd('cancel'));
```

`dd('cancel')` is fired when the modal opens but i want to trigger `dd` after i click on cancel.

**Comment by leandrocfe** [comment]
maybe this
```php
->modalCancelAction(
    \Filament\Actions\StaticAction::make('cancel')
        ->button()
        ->action('cancel')
)
```
in the page file
```php
public function cancel(): void
{
    //do something
}
```

**Reply by bikash-s-bhandari** [reply]
how to do it from table resource? i want to pass record when modal is close
 Tables\Actions\Action::make('Notes')
           
                    ->icon('heroicon-o-pencil-square')
                    ->iconButton()
                    
                    
                    ->modalCancelAction(
                       function(TimeCard $timeCard){
                        $id = $timeCard->id;
                      
                        return \Filament\Actions\StaticAction::make('cancel')
                        ->button()
                        ->action('cancel({{$id}})');
                       }
                    )

**Reply by medentis** [reply]
We are using the theme from https://essentials.pixfort.com/
Unfortunately, the Delightful Downloads plugin contains buggy custom javascript code that removes and overwrites the default global WordPress javascript variables to replace them with custom data in the plugin's javascript code. As a result, the theme unfortunately only works to a very limited extent, as some control panels cannot be opened.
Here is a picture of the code.

I would be very grateful if you could correct the code.

Best regards
Stefan
![344935120-6c9739f8-e790-4391-b068-ec51f2924b41](https://github.com/user-attachments/assets/c5f18879-4a3d-4acc-b15f-f8d5bddd358f)

**Comment by juanborras** [comment]
> Hey! Is it possible to modify the cancel button in the action modal? If I click on cancel, I would like it to trigger something.
> 
> ```
> return Action::make('move')
>     ->requiresConfirmation()
>     ->modalCancelAction(fn() => dd('cancel'));
> ```
> 
> `dd('cancel')` is fired when the modal opens but i want to trigger `dd` after i click on cancel.

Maybe you already solve the issue... for me worked using this info
https://filamentphp.com/docs/3.x/actions/modals#modifying-a-default-modal-footer-action-button

**Comment by pintend** [comment]
```php
->modalCancelAction(function (Action $action, Component $livewire) {
    return $action->livewire($livewire)
        ->close(false)
        ->cancelParentActions(true)
        ->action(function () {
            // Do Something
        });
})
```
